src/connection/host_lookup.c

```diff
@@ -0,0 +1,671 @@
+#endif				/* HAVE_GETHOSTBYNAME_R */
+{
+#if defined (WINDOWS)
+  int ret_val = 0;
+  return 0;
+#endif
+
+  struct hostent *hp_buf = NULL;
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+  int ret_val = 0;
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+  struct hostent *ret_val = NULL;
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+  int ret_val = 0;
+#else
+#error "HAVE_GETHOSTBYNAME_R"
+#endif
+#endif
+
+  if (prm_get_bool_value (PRM_ID_USE_USER_HOSTS) == USE_GLIBC_HOSTS)
+    {
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+      ret_val = gethostbyname_r (name, ret, buf, buflen, result, h_errnop);
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+      ret_val = gethostbyname_r (name, ret, buf, buflen, h_errnop);
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+      ret_val = gethostbyname_r (name, ret, &ht_data);
+#else
+#error "HAVE_GETHOSTBYNAME_R"
+#endif
+#endif
+      goto return_phase;
+    }
+  else
+    {
+      hp_buf = host_lookup_internal (name, NULL, HOSTNAME_TO_IPADDR);
+    }
+
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+  if (hp_buf == NULL)
+    {
+      ret_val = EINVAL;
+
+      goto return_phase;
+    }
+  if (((*result) = (struct hostent *) malloc (sizeof (struct hostent))) == NULL)
+    {
+      ret_val = ENOMEM;
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (struct hostent));
+
+      goto return_phase;
+    }
+  memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
+  memcpy ((void *) *result, (void *) hp_buf, sizeof (struct hostent));
+
+  ret_val = 0;
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+  if (hp_buf == NULL)
+    {
+//err_print fprintf (stdout, msgcat_message (MSGCAT_CATALOG_CUBRID, MSGCAT_SET_ERROR, -ER_FAILED));
+      ret_val = NULL;
+
+      goto return_phase;
+    }
+
+  memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
+
+  ret_val = ret;
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+  if (hp_buf == NULL)
+    {
+//err_print fprintf (stdout, msgcat_message (MSGCAT_CATALOG_CUBRID, MSGCAT_SET_ERROR, -ER_FAILED));
+      ret_val = -1;
+
+      goto return_phase;
+    }
+
+  memcpy ((void *) ret, (void *) hp_buf, sizeof (struct hostent));
+
+  ret_val = 0;
+#else
+#error "HAVE_GETHOSTBYNAME_R"
+#endif /* HAVE_GETHOSTBYNAME_R_GLIBC */
+#endif /* HAVE_GETHOSTBYNAME_R */
+
+return_phase:
+  return ret_val;
+}
+
+
+int
+getnameinfo_uhost (struct sockaddr *addr, socklen_t addrlen, char *host, size_t hostlen, char *serv, size_t servlen,
+		   int flags)
+{
+  int ret;
+  struct hostent *hp = NULL;
+
+  if (prm_get_bool_value (PRM_ID_USE_USER_HOSTS) == USE_GLIBC_HOSTS)
+    {
+      ret = getnameinfo (addr, addrlen, host, hostlen, serv, servlen, flags);
+    }
+  else
+    {
+      if ((hp = host_lookup_internal (NULL, addr, IPADDR_TO_HOSTNAME)) != NULL)
+	{
+	  strncpy (host, hp->h_name, hostlen);
+	  host[CUB_MAXHOSTNAMELEN - 1] = '\0';
+
```

Comment:
You need

`free (hp);`

Reply:
I don't think so.
Because if hp is NULL, we don't need to free the memory.

Reply:
I would not think it is needed to insert free memory.

'hp' is not memory allocation structure and even it is NULL.